### DISCUSSION TITLE
Publish the custom panel to self hosted docker

### DISCUSSION THREAD

**Discussion by BatuhanCakir** [foxglove/community#793] [Q&A] [answered]
Hi,
Is have a problem that the self hosted version doesn't find the custom panel while the normal installed foxglove finds it. Is there a way I can publish it to the self hosted version ?

Thanks in advance

**Comment by defunctzombie** [comment]
Self-hosted docker does not find extensions. That is a feature of our hosted web app (via org provisioned extensions) or the desktop app. The self-hosted app expects you to build everything into the app.

**Reply by BatuhanCakir** [reply]
how can i build the extension into the app ?

**Reply by amacneil** [reply, marked as answer]
It's not currently supported out of the box to load extensions automatically in the self-hosted version. You could modify the studio code to load an extension automatically during boot.

Alternatively, you can publish the extension publicly, or you can publish it internally to your team if you use the official Foxglove build at https://studio.foxglove.dev/

Documentation on your options here: https://foxglove.dev/docs/studio/extensions/publish

**Reply by BatuhanCakir** [reply]
is there a documentation how i can modify the studio code ?. If I want to publish it to the foxglove build I need to have a paid plan right ?. Is there any academic access options ?

**Reply by amacneil** [reply]
We have a free plan that works for up to 3 users. If you are doing academic work and need more, send us a note sales@foxglove.dev and we can help out.
